tests/test_mylog.py

```diff
@@ -1,255 +1,683 @@
+import base64
+import os
 import secrets
+import sys
+from time import time as get_unix_time
+from typing import Callable, Sequence, Tuple, TypeVar, Union
 
 import pytest
+import termcolor
 
 import mylog
 
+T = TypeVar("T")
 
-def test_nolock_enter():
-    nl = mylog.NoLock()
 
-    enter_rv = nl.__enter__()
+def _randint(a: int, b: int) -> int:
+    # random.randint is used only here,
+    # so we have to use "nosec" (vvvvvvvvvvvv) only once
+    return secrets.SystemRandom().randint(a, b)  # nosec B311
 
-    assert isinstance(enter_rv, bool) or enter_rv == 1
 
+def _randbytes(length: int) -> bytes:
+    # random.randbytes is used only here,
+    # so we have to use "nosec" only once
+    return secrets.token_bytes(length)  # nosec B311
 
-def test_nolock_exit():
-    nl = mylog.NoLock()
 
-    exit_rv = nl.__exit__(None, None, None)
+def _randchoice(seq: Sequence[T]) -> T:
+    # random.choice is used only here,
+    # so we have to use "nosec" only once
+    return secrets.SystemRandom().choice(seq)  # nosec B311
 
-    assert exit_rv is None
+
+def _random_int(
+    neg_ok: bool, *, only_if: Callable[[int], bool] = lambda _: True
+) -> int:
+    # sourcery skip: assign-if-exp, reintroduce-else
+    if neg_ok:
+        rv = _randint(-50, 50)
+    rv = _randint(0, 100)
+    if only_if(rv):
+        return rv
+    return _random_int(neg_ok, only_if=only_if)  # pragma: no cover
+
+
+def _random_bytes(
+    *, only_if: Callable[[int], bool] = lambda _: True
+) -> bytes:
+    rv = _randbytes(_random_int(False))
+    if only_if(rv):
+        return rv
+    return _random_bytes(only_if=only_if)  # pragma: no cover
+
+
+def _random_urlsafe(
+    *, only_if: Callable[[int], bool] = lambda _: True
+) -> str:
+    tok = _random_bytes()
+    rv = base64.urlsafe_b64encode(tok).rstrip(b"=").decode("ascii")
+    if only_if(rv):
+        return rv
+    return _random_urlsafe(only_if=only_if)  # pragma: no cover
+
+
+def _random_nonlevel_int(
+    *, only_if: Callable[[int], bool] = lambda _: True
+) -> int:
+    rv = _random_int(
+        True,
+        only_if=lambda num: num not in {10, 20, 30, 40, 50},
+    )
+    if only_if(rv):
+        return rv
+    return _random_nonlevel_int(only_if=only_if)  # pragma: no cover
+
+
+def _random_level() -> Tuple[
+    Union[mylog.Level, int, str], mylog.Level
+]:
+    lvl = _randchoice(tuple(mylog.Level))
+    _to = _randchoice(("lvl", "int", "str"))
+    if _to == "lvl":
+        return lvl, lvl
+    elif _to == "int":
+        return lvl.value, lvl
+    elif _to == "str":
+        return lvl.name, lvl
+    # else is not needed
+
+
+def random_anything(
+    *, only_if: Callable[[int], bool] = lambda _: True
```

Comment:
```suggestion
    *, only_if: Callable[[Union[bytes, int, str]], bool] = lambda _: True
```
`only_if` is called with more than just `int`.